### PR TITLE
feat(i18n): enhanced i18n support

### PR DESCRIPTION
**Pull request by pavelpisarev**
### Context

This PR adds more i18n support

### Changes & Results

* i18n keys for UserPreferencesModal and AboutModal
* ability to customize date and time format in WorkList
* i18n keys for DateRange buttons
* seriesTableColumns keys moved to WorkList from StudyListExpandedRow
* InputLabelWrapper keys for filters moved to filtersMeta
* 'Studies' in StudyListFilter renamed to 'Number of studies'. 'Number of studies' and numOfStudies variable moved in one tag

### Testing

You can try switching to another language.
To change the date and time format, add localDateFormat and localTimeFormat to Common.json (e.g. "localDateFormat": "DD.MM.YYYY",  "localTimeFormat": "HH:mm"). If the key is not specified, the default value is used.

#### PR

- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [x] OS: Windows 11
- [x] Node version: v18.18.0
- [x] Browser: Chrome 118.0.5993.88

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

File: platform/i18n/src/locales/en-US/AboutModal.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
 {
   "Browser": "Browser",
-  "Build Number": "Build Number",
-  "Latest Master Commits": "Latest Master Commits",
+  "Build number": "Build Number",
+  "Last master commits": "Latest Master Commits",
   "More details": "More details",
   "Name": "Name",
   "OHIF Viewer - About": "OHIF Viewer - About",
```
Comment (jbocce) [inline platform/i18n/src/locales/en-US/AboutModal.json:7]: I think with the changes in this PR, this should be...

`"About OHIF Viewer":"About OHIF Viewer"`

File: platform/i18n/src/locales/es/AboutModal.json [hunk 2]
```diff
@@ -1,14 +1,14 @@
 {
   "Browser": "Navegador",
-  "Build Number": "Número de compilación",
-  "Latest Master Commits": "Últimos Master Commits",
+  "Build number": "Número de compilación",
+  "Last master commits": "Últimos Master Commits",
   "More details": "Más detalles",
   "Name": "Nombre",
   "OHIF Viewer - About": "Sobre OHIF Viewer",
```
Comment (jbocce) [inline platform/i18n/src/locales/es/AboutModal.json:7]: The key here should be `"About OHIF Viewer"`

File: platform/i18n/src/locales/pt-BR/AboutModal.json [hunk 3]
```diff
@@ -1,14 +1,14 @@
 {
   "Browser": "Navegador",
-  "Build Number": "Número da compilação",
-  "Latest Master Commits": "Últimos Commits na Master",
+  "Build number": "Número da compilação",
+  "Last master commits": "Últimos Commits na Master",
   "More details": "Mais detalhes",
   "Name": "Nome",
   "OHIF Viewer - About": "OHIF Viewer - Sobre",
```
Comment (jbocce) [inline platform/i18n/src/locales/pt-BR/AboutModal.json:7]: `"About OHIF Viewer"`

File: platform/i18n/src/locales/test-LNG/AboutModal.json [hunk 4]
```diff
@@ -1,14 +1,14 @@
 {
   "Browser": "Browser",
   "Build Number": "Build Number",
-  "Latest Master Commits": "Latest Master Commits",
+  "Last master commits": "Latest Master Commits",
   "More details": "More details",
   "Name": "Name",
   "OHIF Viewer - About": "OHIF Viewer - About",
```
Comment (jbocce) [inline platform/i18n/src/locales/test-LNG/AboutModal.json:7]: `"About OHIF Viewer"` for the key.

File: platform/i18n/src/locales/tr-TR/AboutModal.json [hunk 5]
```diff
@@ -1,14 +1,14 @@
 {
   "Browser": "Tarayıcı",
-  "Build Number": "Derleme Numarası",
-  "Latest Master Commits": "Son Kaynak Kod Güncellemesi",
+  "Build number": "Derleme Numarası",
+  "Last master commits": "Son Kaynak Kod Güncellemesi",
   "More details": "Daha Fazla Detay",
   "Name": "İsim",
   "OHIF Viewer - About": "OHIF Viewer - Hakkında",
```
Comment (jbocce) [inline platform/i18n/src/locales/tr-TR/AboutModal.json:7]: `"About OHIF Viewer"` for the key.

File: platform/i18n/src/locales/de/StudyList.json
Comment (jbocce) [inline platform/i18n/src/locales/de/StudyList.json:1]: Should we add `"Series"` here?

File: platform/app/src/routes/WorkList/filtersMeta.js [hunk 6]
```diff
@@ -1,35 +1,37 @@
...
-    displayName: 'StudyDate',
+    displayName: i18n.t('StudyList:StudyDate'),
     inputType: 'DateRange',
     isSortable: true,
     gridCol: 5,
   },
   {
     name: 'description',
-    displayName: 'StudyDescription',
+    displayName: i18n.t('StudyList:StudyDescription'),
```
Comment (jbocce) [inline platform/app/src/routes/WorkList/filtersMeta.js:27]: In Worklist.tsx we used `'StudyList:Description'`, so I think here we want that here too, no?

File: platform/ui/src/components/AboutModal/AboutModal.tsx [hunk 7]
```diff
@@ -73,71 +75,71 @@ const AboutModal = ({ buildNumber, versionNumber, commitHash }) => {
   );
   return (
     <div>
-      {renderRowTitle('Important Links')}
+      {renderRowTitle(t('Important links'))}
```
Comment (jbocce) [inline platform/ui/src/components/AboutModal/AboutModal.tsx:78]: Should we add `'Important links'` to some of the various AboutModal.json files? At least for en-US and test-LNG maybe?

File: platform/ui/src/components/AboutModal/AboutModal.tsx [hunk 8]
```diff
@@ -73,71 +75,71 @@ const AboutModal = ({ buildNumber, versionNumber, commitHash }) => {
...
       <div className="flex flex-col">
         <Row
-          title="Repository URL"
+          title={t('Repository URL')}
           value="https://github.com/OHIF/Viewers/"
           link="https://github.com/OHIF/Viewers/"
         />
         <Row
-          title="Data Citation"
+          title={t('Data citation')}
```
Comment (jbocce) [inline platform/ui/src/components/AboutModal/AboutModal.tsx:112]: Should we add this to various AboutModal.json files?

File: platform/ui/src/components/AboutModal/AboutModal.tsx [hunk 9]
```diff
@@ -73,71 +75,71 @@ const AboutModal = ({ buildNumber, versionNumber, commitHash }) => {
...
         />
         {/* <Row
-          title="Last Master Commits"
+          title={t('Last master commits')}
           value="https://github.com/OHIF/Viewers/"
           link="https://github.com/OHIF/Viewers/"
         /> */}
         <Row
-          title="Version number"
+          title={t('Version number')}
```
Comment (jbocce) [inline platform/ui/src/components/AboutModal/AboutModal.tsx:122]: Add to AboutModal.json files?

File: platform/ui/src/components/AboutModal/AboutModal.tsx [hunk 10]
```diff
@@ -73,71 +75,71 @@ const AboutModal = ({ buildNumber, versionNumber, commitHash }) => {
...
           <Row
-            title="Build number"
+            title={t('Build number')}
             value={buildNumber}
           />
         )}
         {commitHash && (
           <Row
-            title="Commit Hash"
+            title={t('Commit hash')}
```
Comment (jbocce) [inline platform/ui/src/components/AboutModal/AboutModal.tsx:133]: Add to AboutModal.json files?

File: platform/ui/src/components/DateRange/DateRange.tsx [hunk 11]
```diff
@@ -149,11 +150,11 @@ const DateRange = props => {
...
       renderCalendarInfo={renderDatePresets}
       renderMonthElement={renderMonthElement}
-      startDatePlaceholderText={'Start Date'}
-      endDatePlaceholderText={'End Date'}
+      startDatePlaceholderText={t('Start Date')}
+      endDatePlaceholderText={t('End Date')}
       phrases={{
-        closeDatePicker: 'Close',
-        clearDates: 'Clear dates',
+        closeDatePicker: t('Close'),
```
Comment (jbocce) [inline platform/ui/src/components/DateRange/DateRange.tsx:156]: Add to DatePicker.json files or perhaps change to `t('Common:Close')`?

File: platform/ui/src/components/DateRange/DateRange.tsx [hunk 12]
```diff
@@ -52,6 +32,27 @@ const DateRange = props => {
   const { id, onChange, startDate, endDate } = props;
   const [focusedInput, setFocusedInput] = useState(null);
   const renderYearsOptionsCallback = useCallback(renderYearsOptions, []);
+  const { t } = useTranslation('DatePicker');
+  const today = moment();
+  const lastWeek = moment().subtract(7, 'day');
+  const lastMonth = moment().subtract(1, 'month');
+  const studyDatePresets = [
+    {
```
Comment (jbocce) [inline platform/ui/src/components/DateRange/DateRange.tsx:40]: Should we add `'Today'`, `'Last 7 days'` and `'Last 30 days'` to the various DatePicker.json files? Again, just the ones you are comfortable with.

File: platform/ui/src/components/HotkeysPreferences/HotkeysPreferences.tsx [hunk 13]
```diff
@@ -30,7 +30,7 @@ const HotkeysPreferences = ({
   const splitedHotkeys = splitHotkeyDefinitionsAndCreateTuples(visibleHotkeys);
 
   if (!Object.keys(hotkeyDefinitions).length) {
-    return 'No hotkey definitions';
+    return t('No hotkey definitions');
```
Comment (jbocce) [inline platform/ui/src/components/HotkeysPreferences/HotkeysPreferences.tsx:33]: Should we add this to the UserPreferencesModal.json files that you are comfortable with?

File: platform/ui/src/components/StudyListFilter/StudyListFilter.tsx [hunk 14]
```diff
@@ -67,18 +67,11 @@ const StudyListFilter = ({
...
-                  data-cy={'num-studies'}
-                >
-                  {numOfStudies > 100 ? '>100' : numOfStudies}
-                </Typography>
                 <Typography
                   variant="h6"
                   className="text-primary-light self-end pb-1"
                 >
-                  {t('Studies')}
+                  {`${t('Number of studies')}: ${numOfStudies > 100 ? '>100' : numOfStudies}`}
```
Comment (jbocce) [inline platform/ui/src/components/StudyListFilter/StudyListFilter.tsx:74]: Very good change for language neutrality. Thank you.

**Review by jbocce — CHANGES_REQUESTED** [review]
Thanks for these great changes. Please see my comments.

File: platform/app/src/routes/WorkList/filtersMeta.js (hunk 6 above)
Comment (pavelpisarev) [reply platform/app/src/routes/WorkList/filtersMeta.js:27]: Didn't notice the "Description" key. Thank you!

File: platform/ui/src/components/HotkeysPreferences/HotkeysPreferences.tsx (hunk 13 above)
Comment (pavelpisarev) [reply platform/ui/src/components/HotkeysPreferences/HotkeysPreferences.tsx:33]: There is already existing "No hotkeys found" key in "User Preferences Modal". I'll fix it in the next commits

**Comment by pavelpisarev** [comment]
@jbocce thanks for your review! I plan to continue working on localization, but there is a problem. Some keys are incorrectly named, some are simply not used.

Do you know if it is possible to regenerate all used keys using locize.io or some other library in the project? For example, regenerate TEST-LNG and use it as a template.

**Comment by pavelpisarev** [comment]
> When you say that you will continue working on localization, do you mean in this PR or in another? I think it might be best in another PR, but let me know your thoughts. Also I am not aware of a way to regenerate all of the keys, but I will ask around to try and find out.

I meant another PR. This one is complete

File: platform/i18n/src/locales/en-US/DatePicker.json [hunk 15]
```diff
@@ -1,5 +1,9 @@
 {
   "Clear dates": "Clear dates",
+  "Close": "$t(Common:Close)",
```
Comment (jbocce) [inline platform/i18n/src/locales/en-US/DatePicker.json:3]: Should we add this to every DatePicker.json file?

**Review by jbocce — CHANGES_REQUESTED** [review]
Thanks for all the updates. Just one more to consider. Thanks.

File: platform/i18n/src/locales/en-US/DatePicker.json (hunk 15 above)
Comment (pavelpisarev) [reply platform/i18n/src/locales/en-US/DatePicker.json:3]: I think that test-LNG and en locales are enough. In some localizations there is either no "Close" key or a Common.json file.
So I would like to do test-LNG and en first and then I can make a template for other languages. What do you think?

Comment (jbocce) [reply platform/i18n/src/locales/en-US/DatePicker.json:3]: Yes. That makes sense.

**Review by jbocce — APPROVED** [review]
Thanks for all these changes. Much appreciated!

**Comment by pavelpisarev** [comment]
@jbocce Thank you very much for your review! Will someone do a merge or do I have rights to do this? Never done this on github, sorry.

**Comment by jbocce** [comment]
> @jbocce Thank you very much for your review! Will someone do a merge or do I have rights to do this? I don’t know what the workflow is here.

You are very welcome. I believe @sedghi will have a final look and do the merge.

**Comment by sedghi** [comment]
thanks for your PR, can you resolve conflict? before I merge

**Pull request merged**